internal/parse/parse.go

```diff
@@ -330,8 +331,8 @@ func GetCacheMount(args []string, store storage.Store, imageMountLabel string, a
 		// create a common cache directory, which persists on hosts within temp lifecycle
 		// add subdirectory if specified
 
-		// cache parent directory
-		cacheParent := filepath.Join(internalUtil.GetTempDir(), BuildahCacheDir)
+		// cache parent directory: creates eperate cache parent for each user.
+		cacheParent := filepath.Join(internalUtil.GetTempDir(), filepath.Join(BuildahCacheDir, strconv.Itoa(unshare.GetRootlessUID())))
```

Comment:
The problem with this is the first user to do it will still own `BuildahCacheDir` and the next user won't be able to create their subfolder under it. This should work, though:

```suggestion
		cacheParent := filepath.Join(internalUtil.GetTempDir(), BuildahCacheDir+"-"+strconv.Itoa(unshare.GetRootlessUID()))
```

I'll see if I can get it tested today.